### PR TITLE
Updated batt_smbus. Expanded battery_status.msg. Fixed mavlink_messages.cpp temperature.

### PR DESCRIPTION
**Pull request by AlexKlimaj**
Updated and expanded batt_smbus to work with bq40z50-R2. Expanded battery_status.msg. Fixed mavlink_messages.cpp temperature, added commented out expanded battery_status.msg parameters for future mavlink expansion.

I've been flying this code with our smart battery for a couple of weeks now without issue. 

Planning future enhancements to fall back on ADC battery driver if communication with the smart battery fails for some reason. 

Batt_source needs to be set to 1 to bypass battery.cpp ADC driver.

[QGC Mavlink Inspector](https://imgur.com/D4zkFbk)

[QGC Battery Status](https://imgur.com/a/n7qNT)

Nut Shell Output:
> nsh> batt_smbus stop
nsh>
nsh>
nsh> batt_smbus start
batt_smbus on I2C bus 1 at 0x0b (bus: 100 KHz, max: 100 KHz)
INFO  [batt_smbus] battery found at 0xb
INFO  [batt_smbus] smart battery connected
nsh>
nsh>
nsh> batt_smbus test
INFO  [batt_smbus] V=16.53 C=0.34 AveC=0.35 DismAh=0 Cap:2552 TempC:26.45 Remaining:0.96
 RunTimeToEmpty:437 AveTimeToEmpty:427 CycleCount:1 SerialNum:7078
INFO  [batt_smbus] V=16.53 C=0.33 AveC=0.35 DismAh=0 Cap:2552 TempC:26.45 Remaining:0.96
 RunTimeToEmpty:437 AveTimeToEmpty:427 CycleCount:1 SerialNum:7078
INFO  [batt_smbus] V=16.53 C=0.33 AveC=0.35 DismAh=0 Cap:2552 TempC:26.45 Remaining:0.96
 RunTimeToEmpty:437 AveTimeToEmpty:427 CycleCount:1 SerialNum:7078
INFO  [batt_smbus] V=16.53 C=0.33 AveC=0.35 DismAh=0 Cap:2552 TempC:26.45 Remaining:0.96
 RunTimeToEmpty:437 AveTimeToEmpty:427 CycleCount:1 SerialNum:7078
INFO  [batt_smbus] V=16.53 C=0.33 AveC=0.35 DismAh=0 Cap:2552 TempC:26.45 Remaining:0.96
 RunTimeToEmpty:437 AveTimeToEmpty:427 CycleCount:1 SerialNum:7078
INFO  [batt_smbus] V=16.53 C=0.32 AveC=0.34 DismAh=0 Cap:2552 TempC:26.45 Remaining:0.96
 RunTimeToEmpty:457 AveTimeToEmpty:428 CycleCount:1 SerialNum:7078
INFO  [batt_smbus] V=16.53 C=0.32 AveC=0.34 DismAh=0 Cap:2552 TempC:26.45 Remaining:0.96
 RunTimeToEmpty:457 AveTimeToEmpty:428 CycleCount:1 SerialNum:7078
INFO  [batt_smbus] V=16.53 C=0.32 AveC=0.34 DismAh=0 Cap:2552 TempC:26.45 Remaining:0.96
 RunTimeToEmpty:457 AveTimeToEmpty:428 CycleCount:1 SerialNum:7078
INFO  [batt_smbus] V=16.53 C=0.32 AveC=0.34 DismAh=0 Cap:2552 TempC:26.45 Remaining:0.96
 RunTimeToEmpty:457 AveTimeToEmpty:428 CycleCount:1 SerialNum:7078
INFO  [batt_smbus] V=16.53 C=0.32 AveC=0.34 DismAh=0 Cap:2552 TempC:26.45 Remaining:0.96
 RunTimeToEmpty:457 AveTimeToEmpty:428 CycleCount:1 SerialNum:7078
INFO  [batt_smbus] V=16.53 C=0.33 AveC=0.34 DismAh=0 Cap:2552 TempC:26.45 Remaining:0.96
 RunTimeToEmpty:452 AveTimeToEmpty:429 CycleCount:1 SerialNum:7078
INFO  [batt_smbus] V=16.53 C=0.33 AveC=0.34 DismAh=0 Cap:2552 TempC:26.45 Remaining:0.96
 RunTimeToEmpty:452 AveTimeToEmpty:429 CycleCount:1 SerialNum:7078
INFO  [batt_smbus] V=16.53 C=0.34 AveC=0.34 DismAh=0 Cap:2552 TempC:26.45 Remaining:0.96
 RunTimeToEmpty:452 AveTimeToEmpty:429 CycleCount:1 SerialNum:7078
INFO  [batt_smbus] V=16.53 C=0.34 AveC=0.34 DismAh=0 Cap:2552 TempC:26.45 Remaining:0.96
 RunTimeToEmpty:452 AveTimeToEmpty:429 CycleCount:1 SerialNum:7078
nsh>
nsh> listener battery_status
TOPIC: battery_status instance 0 #1
timestamp: 640048256
voltage_v:  16.5170
voltage_filtered_v:  16.5170
current_a:   0.3220
current_filtered_a:   0.3220
average_current_a:   0.3440
discharged_mah:  10.0000
remaining:   0.9581
scale:   0.0000
temperature:  26.8500
cell_count: 0
connected: True
system_source: False
priority: 0
capacity: 2552
cycle_count: 1
run_time_to_empty: 454
average_time_to_empty: 426
serial_number: 28792
is_powering_off: False
warning: 0
nsh>

### CODE REVIEW

File: src/drivers/batt_smbus/batt_smbus.cpp [hunk 1]
```diff
@@ -974,40 +880,37 @@ batt_smbus_main(int argc, char *argv[])
...
+		return 1;
 	}
 
 	const char *verb = argv[optind];
 
 	if (!strcmp(verb, "start")) {
 		if (g_batt_smbus != nullptr) {
-			PX4_ERR("already started");
-			return 1;
+			errx(1, "already started");
```
Comment (dagar) [inline src/drivers/batt_smbus/batt_smbus.cpp]: Can you remove anything that actually exists? For example replace errx with PX4_ERR and a return. Then this driver will be usable out of the box on linux where an exit actually kills the px4 process.

Comment (AlexKlimaj) [reply src/drivers/batt_smbus/batt_smbus.cpp]: Done

File: src/modules/mavlink/mavlink_messages.cpp [hunk 2]
```diff
@@ -573,7 +573,13 @@ class MavlinkStreamSysStatus : public MavlinkStream
 			bat_msg.energy_consumed = -1;
 			bat_msg.current_battery = (battery_status.connected) ? battery_status.current_filtered_a * 100 : -1;
 			bat_msg.battery_remaining = (battery_status.connected) ? ceilf(battery_status.remaining * 100.0f) : -1;
-			bat_msg.temperature = INT16_MAX;
+			bat_msg.temperature = (battery_status.connected) ? (int16_t)battery_status.temperature : INT16_MAX;
+			//bat_msg.average_current_battery = (battery_status.connected) ? battery_status.average_current_a * 100.0f : -1;
```
Comment (dagar) [inline src/modules/mavlink/mavlink_messages.cpp]: Why is this commented?

Comment (AlexKlimaj) [reply src/modules/mavlink/mavlink_messages.cpp]: It needs an updated version of Mavlink with these messages to build correctly. I put them there for the future.

**Comment by dagar** [comment]
@LorenzMeier do you have anything else we could test this with? Should we make sure it fully works on a Solo?

**Comment by LorenzMeier** [comment]
Same question as in the last PR: Why did you remove functionality to read some values from SMBus? This PR is not an update, it is a change to one particular power management part and I'd like to understand the rationale.

**Comment by AlexKlimaj** [comment]
@LorenzMeier 

I removed functions that don't make much sense for a smart battery. The values being read also weren't being used or passed anywhere. 

Also, the code didn't build as of a few weeks ago, so it's safe to say nobody has been using it recently. 

`uint8_t     device_name(uint8_t *dev_name, uint8_t max_length);` - Why do you need to get the name of the battery?

`uint8_t     device_chemistry(uint8_t *dev_chem, uint8_t max_length);` - This is hardcoded to MAV_BATTERY_TYPE_LIPO in mavlink_messages.cpp. 

`bool is_solo_battery();` - Why is this needed?

`_button_press_counts(0)` - Fuel gauges I've seen don't report button presses natively. Why would you need this?

**Comment by LorenzMeier** [comment]
That makes sense. I just wanted to be sure the code matches the considerations and intentions, which it does.

**Comment by mcsauder** [comment]
Thanks @LorenzMeier  and @dagar  for your review (and also acceptance) of this code!

We've put another ~1.5hrs / 8 flight packs through flight testing today as well as the full day of bench evaluation on ~10 fuel gauge equipped packs with today's Firmware/master branch... flawless results so far.  It is very fun to breathe new life into the smbus code!  

Thanks again for your careful consideration.

-Mark

**Pull request merged**